Components/ImportService.php

```diff
@@ -240,27 +256,35 @@ public function importRemoteCategory($localCategoryId, $remoteCategoryKey, $remo
+        $categories = $this->autoCategoryResolver->convertTreeToKeys($remoteCategoryNodes, $localCategory->getId(), false);
 
         foreach ($categories as $category) {
-            $articleIds = $this->productToRemoteCategoryRepository->findArticleIdsByRemoteCategory($category['categoryKey']);
-
-            while ($currentIdBatch = array_splice($articleIds, 0, 10)) {
-                $articles = $this->articleRepository->findBy(['id' => $currentIdBatch]);
-                /** @var \Shopware\Models\Article\Article $article */
-                foreach ($articles as $article) {
-                    /** @var \Shopware\Models\Category\Category $categoryModel */
-                    $categoryModel = $category['model'];
-                    if ($article->getCategories()->contains($categoryModel->getParent())) {
-                        $article->removeCategory($categoryModel->getParent());
-                    }
-                    $article->addCategory($category['model']);
-                    $attribute = $article->getAttribute();
-                    $attribute->setConnectMappedCategory(true);
-                    $this->manager->persist($article);
-                    $this->manager->persist($attribute);
-                }
-                $this->manager->flush();
+            $articleIds = $this->productToRemoteCategoryRepository->findArticleIdsByRemoteCategory($category['remoteCategory']);
+            foreach ($articleIds as $articleId) {
+                $this->categoryDenormalization->addAssignment($articleId, $category['categoryKey']);
+                $this->categoryDenormalization->removeAssignment($articleId, $category['parentId']);
+                $this->manager->getConnection()->executeQuery(
+                    'INSERT IGNORE INTO `s_articles_categories` (`articleID`, `categoryID`) VALUES (?, ?)',
+                    [$articleId,  $category['categoryKey']]
+                );
+                $this->manager->getConnection()->executeQuery(
+                    'DELETE FROM `s_articles_categories` WHERE `articleID` = :articleID AND `categoryID` = :categoryID',
+                    [
+                        ':articleID' => $articleId,
+                        ':categoryID' => $category['parentId']
+                    ]
+                );
+                $detailId = $this->manager->getConnection()->fetchColumn(
+                    'SELECT id FROM `s_articles_details` WHERE `articleID` = :articleID',
+                    ['articleID' => $articleId]
+                );
+                $this->manager->getConnection()->executeQuery(
+                    'INSERT  INTO `s_articles_attributes` (`articleID`, `articledetailsID`, `connect_mapped_category`) 
+                        VALUES (?, ?, 1)
+                        ON DUPLICATE KEY UPDATE `connect_mapped_category` = 1
+                    ',
+                    [$articleId,  $detailId]
```

Comment:
will work only with articleId I think

Reply:
no the tests fail then 
serviceContainerTest::testImportCategory